components/inputs/input-number.js

```diff
@@ -1,20 +1,52 @@
 import './input-text.js';
 import { css, html, LitElement } from 'lit-element/lit-element.js';
-import { formatNumber, parseNumber } from '@brightspace-ui/intl/lib/number.js';
+import { formatNumber, getNumberDescriptor, parseNumber } from '@brightspace-ui/intl/lib/number.js';
 import { FormElementMixin } from '../form/form-element-mixin.js';
 import { getUniqueId } from '../../helpers/uniqueId.js';
 import { ifDefined } from 'lit-html/directives/if-defined.js';
 import { LocalizeCoreElement } from '../../lang/localize-core-element.js';
 import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
 
-function formatValue(value, minFractionDigits, maxFractionDigits) {
+function formatValue(value, minFractionDigits, maxFractionDigits, showTrailingZeroes, numDecimalDigits) {
+	let adjustedMinFractionDigits = minFractionDigits;
+	if (showTrailingZeroes && numDecimalDigits > 0) {
+		adjustedMinFractionDigits = Math.max(minFractionDigits || 0, numDecimalDigits);
+		if (maxFractionDigits !== undefined) {
+			adjustedMinFractionDigits = Math.min(
+				adjustedMinFractionDigits,
+				maxFractionDigits
+			);
+		}
+	}
 	const options = {
 		maximumFractionDigits: maxFractionDigits,
-		minimumFractionDigits: minFractionDigits
+		minimumFractionDigits: adjustedMinFractionDigits
 	};
 	return formatNumber(value, options);
 }
 
+function countDecimalDigits(value) {
```

Comment:
This counts the number of digits beyond the decimal separator.

Reply:
Okay cool, I thought this was potentially running on formatted user input values. I didn't realize you have to pass `notation: 'scientific'` to NumberFormat for it to return the `e+` format or in this case `Exx`